src/arrays/constructors.jl

```diff
@@ -16,4 +16,17 @@
     one_at_ind!(arr, i) = setindex!(arr, one(eltype(arr)), i)
     single_coeff(i, lens...) = one_at_ind!(zeros(lens), i)
 
-export statevec
\ No newline at end of file
+function coherentstatevec(n::Int,alpha::Number;method="direct")
+    s = zeros(n)
```

Comment:
We now have the type dependent zero array.

I have changed the function to `coherentstatevec_inf` and have removed the use of keyword.

I have added references to the implementations. But yeah I still need to add references to the implementations from the previous pulls. I guess we could have a separate PR for that.